homeworks/03-data-structure/src/binary-tree/binaryTree.ts

```diff
@@ -0,0 +1,69 @@
+export type TreeNode<T> = {
+  value: T;
+  left: TreeNode<T> | null;
+  right: TreeNode<T> | null;
+};
+
+export enum TraverseType {
+  DFS_INORDER = 'DFS_INORDER',
+  DFS_PREORDER = 'DFS_PREORDER',
+  DFS_POSTORDER = 'DFS_POSTORDER',
+  BFS = 'BFS',
+}
+
+export interface Tree<T> {
+  setTreeNode(treeNode: TreeNode<T>): this;
```

Comment:
`constructor` is absent.

Reply:
Sorry, didn't get it. Constructor in interface? Can't understand how to use it here..

Reply:
Sorry for misleading you. Nowadays, still no good way to define the `constructor` type. Never mind this comment.